File: Theory.java
import java.util.*;

public abstract class Theory
{
	protected Random random;
	public abstract double getBeatResolution();
	public abstract CellState[][] initialize();
	public abstract void evolve(CellState[][] songCells);
	public abstract NoteName getNoteName(int pitchNumber);
	public abstract int getOctave(int pitchNumber);
	
	public Theory(Random random)
	{
		this.random = random;
	}
}

File: Automatone/Automatone/Theory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using MoodSwingCoreComponents;

namespace Automatone
{
    public abstract class Theory
    {
        protected static MSRandom random;
        public abstract double getBeatResolution();
        public abstract CellState[,] initialize();
        public abstract void Evolve(CellState[,] songCells);
        public abstract NoteName getNoteName(int pitchNumber);
        public abstract int getOctave(int pitchNumber);

        public Theory(MSRandom rand)
        {
            random = rand;
        }
    }
}
